contrib/scripts/export-creds.sh

```diff
@@ -0,0 +1,106 @@
+
+	profile
+		Profile from aws-vault to export to ~/.aws/credentials. The profile name used in the
+		credentials file will be "${profile}-creds".
+
+Overview
+
+	This script will export credentials in aws-vault for the given profile to ~/.aws/credentials. By
+	default these are temporary credentials created by aws-vault either through get-session-token or
+	assume-role, if --no-session is passed then the long-term credentials or exported instead if
+	possible (this will not work as expected if source_profile is used).
+
+Security
+
+	Generally using temporary credentials through the credentials file is fairly safe. While it may seem
+	odd to write credentials back out to this file when using aws-vault the primary benefit of using 
+	aws-vault is to protect the long-term key while aws-vault is not in use.
+
+	Assuming this script is not run with --no-session then this still applies, only short-term
+	credentials will be present on disk. Having these on disk instead of in environment variables
+	or available through the metadata service can to some degree protect against compromised low
+	privileged users on the same computer, environment variables are typically readable by any
+	user and network access is generally not restricted based on UID. Currently, this script doesn'"'"'t
+	protect against this kind of attack when it is running due to the awscli not supporting secrets
+	passed in through Stdin (TODO: patch to awscli to fix this). This however does not apply after
+	the secret is exported to the credentials file.
+
+	This script also checks the permissions on ~/.aws/credentials to ensure it is not readable by
+	all users on the system and exits with an error if it is.
+'
+}
+
+NO_SESSION=''
+POSITIONAL=()
+while test "$#" -gt 0; do
+    case "$1" in
+        --no-session)
+        NO_SESSION='--no-session'
+        shift
+        ;;
+        -h|--help)
+        help_long
+        exit 1
+        shift
+        ;;
+        *)    # unknown option
+        POSITIONAL+=("$1") # save it in an array for later
+        shift # past argument
+        ;;
+    esac
+done
+
+set -- "${POSITIONAL[@]:-}" # restore positional parameters
+
+if test "$#" -gt 1; then
+    help
+    exit 2
+fi
+
+PROFILE="${1:-}"
+
+if test -z "$PROFILE"; then
+    echo '[ERROR] Must include profile name to export.'
+    help
+    exit 4
+fi
+    
+cred_file="$HOME/.aws/credentials" 
+if ls -l "$cred_file" | grep -q '^.......r..'; then
+    echo "[ERROR] The credentials file '${cred_file}' is readable by other users"
+    exit 101
+fi
+
+
+aws-vault exec $NO_SESSION "${PROFILE}" -- sh -c "
+    set -eu
+    aws configure --profile "${PROFILE}-creds" set region \"\$AWS_REGION\"
+    aws configure --profile "${PROFILE}-creds" set aws_access_key_id \"\$AWS_ACCESS_KEY_ID\"
+    aws configure --profile "${PROFILE}-creds" set aws_secret_access_key \"\$AWS_SECRET_ACCESS_KEY\"
+    aws configure --profile "${PROFILE}-creds" set aws_session_token \"\${AWS_SESSION_TOKEN:-}\"
```

Comment:
This script may be unnecessarily verbose. Why not something simple like the following. 

```bash
#!/bin/sh
# Configure aws-cli using the AWS env vars present
set -eux
aws configure --profile "${1:-$AWS_VAULT}" set region "$AWS_REGION"
aws configure --profile "${1:-$AWS_VAULT}" set aws_access_key_id "$AWS_ACCESS_KEY_ID"
aws configure --profile "${1:-$AWS_VAULT}" set aws_secret_access_key "$AWS_SECRET_ACCESS_KEY"
aws configure --profile "${1:-$AWS_VAULT}" set aws_session_token "${AWS_SESSION_TOKEN:-}"
```

Call it with `aws-vault exec myprofile -- aws-configure-set-via-env-vars.sh myprofile-creds`

What am I missing?

Reply:
This does feel like an elegant solution. My only change would be to drop the `x` from `set -eux` so it doesn't echo credentials to the console. I imagine a script like this will used in a continuous-delivery / GitOps automated deployment scope where output is usually logged.